### DISCUSSION TITLE
Create First App Tutorial 2 Fails to Build

### DISCUSSION THREAD

**Discussion by elibroftw** [unoplatform/uno#11318] [Q&A]
There were times where I had to guess where to paste the code provided, but even then, I got some bugs. Including the fact that the logging package used by default for the templates is 5.0.0 instead of 7.0.0.

https://platform.uno/docs/articles/getting-started-tutorial-2.html#setting-up-our-page

I'm going to test out the first tutorial now.

<details><summary>Error list</summary>

Severity	Code	Description	Project	File	Line	Suppression State
Error	UXAML0001	Unable to find member [Type] on type [IssueItem]	BugTracker	C:\Users\maste\Documents\GitHub\BugTracker\BugTracker\BugTracker\CSC	1	
Warning	WMC1006	Cannot resolve Assembly or Windows Metadata file 'C:\Users\maste\Documents\GitHub\BugTracker\BugTracker\BugTracker\bin\Debug\net7.0-windows10.0.18362\BugTracker.dll'	BugTracker.Windows	C:\Users\maste\Documents\GitHub\BugTracker\BugTracker\BugTracker.Windows\BugTracker.Windows.csproj	1	
Error		This version of WindowsAppSDK requires .NET 6+ and WinRT.Runtime.dll version 2.0 or greater.
    Please update to .NET SDK 6.0.109, 6.0.304 or 6.0.401 (or later).
    Or add a temporary Microsoft.Windows.SDK.NET.Ref reference of version 10.0.*.27 or later.
    For example,
        <PropertyGroup>
            <WindowsSdkPackageVersion>10.0.<Target Windows SDK Build Number>.27</WindowsSdkPackageVersion>
        </PropertyGroup>
	BugTracker	C:\Users\maste\.nuget\packages\microsoft.windowsappsdk\1.2.230118.102\buildTransitive\Microsoft.WindowsAppSDK.targets	59	
Warning	MSB3277	Found conflicts between different versions of "WinRT.Runtime" that could not be resolved.
There was a conflict between "WinRT.Runtime, Version=1.6.0.0, Culture=neutral, PublicKeyToken=99ea127f02d97709" and "WinRT.Runtime, Version=2.0.0.0, Culture=neutral, PublicKeyToken=99ea127f02d97709".
    "WinRT.Runtime, Version=1.6.0.0, Culture=neutral, PublicKeyToken=99ea127f02d97709" was chosen because it was primary and "WinRT.Runtime, Version=2.0.0.0, Culture=neutral, PublicKeyToken=99ea127f02d97709" was not.
    References which depend on "WinRT.Runtime, Version=1.6.0.0, Culture=neutral, PublicKeyToken=99ea127f02d97709" [C:\Users\maste\.nuget\packages\microsoft.windows.sdk.net.ref\10.0.22000.24\lib\WinRT.Runtime.dll].
        C:\Users\maste\.nuget\packages\microsoft.windows.sdk.net.ref\10.0.22000.24\lib\WinRT.Runtime.dll
          Project file item includes which caused reference "C:\Users\maste\.nuget\packages\microsoft.windows.sdk.net.ref\10.0.22000.24\lib\WinRT.Runtime.dll".
            C:\Users\maste\.nuget\packages\microsoft.windows.sdk.net.ref\10.0.22000.24\lib/WinRT.Runtime.dll
    References which depend on "WinRT.Runtime, Version=2.0.0.0, Culture=neutral, PublicKeyToken=99ea127f02d97709" [].
        C:\Users\maste\.nuget\packages\microsoft.windowsappsdk\1.2.230118.102\lib\net6.0-windows10.0.18362.0\Microsoft.InteractiveExperiences.Projection.dll
          Project file item includes which caused reference "C:\Users\maste\.nuget\packages\microsoft.windowsappsdk\1.2.230118.102\lib\net6.0-windows10.0.18362.0\Microsoft.InteractiveExperiences.Projection.dll".
            C:\Users\maste\.nuget\packages\microsoft.windowsappsdk\1.2.230118.102\lib\net6.0-windows10.0.18362.0\Microsoft.InteractiveExperiences.Projection.dll
            C:\Users\maste\.nuget\packages\microsoft.windowsappsdk\1.2.230118.102\lib\net6.0-windows10.0.18362.0\Microsoft.WinUI.dll
        C:\Users\maste\.nuget\packages\microsoft.windowsappsdk\1.2.230118.102\lib\net6.0-windows10.0.18362.0\Microsoft.WinUI.dll
          Project file item includes which caused reference "C:\Users\maste\.nuget\packages\microsoft.windowsappsdk\1.2.230118.102\lib\net6.0-windows10.0.18362.0\Microsoft.WinUI.dll".
            C:\Users\maste\.nuget\packages\microsoft.windowsappsdk\1.2.230118.102\lib\net6.0-windows10.0.18362.0\Microsoft.WinUI.dll
        C:\Users\maste\.nuget\packages\microsoft.windowsappsdk\1.2.230118.102\lib\net6.0-windows10.0.18362.0\Microsoft.Windows.AppLifecycle.Projection.dll
          Project file item includes which caused reference "C:\Users\maste\.nuget\packages\microsoft.windowsappsdk\1.2.230118.102\lib\net6.0-windows10.0.18362.0\Microsoft.Windows.AppLifecycle.Projection.dll".
            C:\Users\maste\.nuget\packages\microsoft.windowsappsdk\1.2.230118.102\lib\net6.0-windows10.0.18362.0\Microsoft.Windows.AppLifecycle.Projection.dll
        C:\Users\maste\.nuget\packages\microsoft.windowsappsdk\1.2.230118.102\lib\net6.0-windows10.0.18362.0\Microsoft.Windows.AppNotifications.Builder.Projection.dll
          Project file item includes which caused reference "C:\Users\maste\.nuget\packages\microsoft.windowsappsdk\1.2.230118.102\lib\net6.0-windows10.0.18362.0\Microsoft.Windows.AppNotifications.Builder.Projection.dll".
            C:\Users\maste\.nuget\packages\microsoft.windowsappsdk\1.2.230118.102\lib\net6.0-windows10.0.18362.0\Microsoft.Windows.AppNotifications.Builder.Projection.dll
        C:\Users\maste\.nuget\packages\microsoft.windowsappsdk\1.2.230118.102\lib\net6.0-windows10.0.18362.0\Microsoft.Windows.AppNotifications.Projection.dll
          Project file item includes which caused reference "C:\Users\maste\.nuget\packages\microsoft.windowsappsdk\1.2.230118.102\lib\net6.0-windows10.0.18362.0\Microsoft.Windows.AppNotifications.Projection.dll".
            C:\Users\maste\.nuget\packages\microsoft.windowsappsdk\1.2.230118.102\lib\net6.0-windows10.0.18362.0\Microsoft.Windows.AppNotifications.Projection.dll
            C:\Users\maste\.nuget\packages\microsoft.windowsappsdk\1.2.230118.102\lib\net6.0-windows10.0.18362.0\Microsoft.Windows.AppNotifications.Builder.Projection.dll
        C:\Users\maste\.nuget\packages\microsoft.windowsappsdk\1.2.230118.102\lib\net6.0-windows10.0.18362.0\Microsoft.Windows.ApplicationModel.DynamicDependency.Projection.dll
          Project file item includes which caused reference "C:\Users\maste\.nuget\packages\microsoft.windowsappsdk\1.2.230118.102\lib\net6.0-windows10.0.18362.0\Microsoft.Windows.ApplicationModel.DynamicDependency.Projection.dll".
            C:\Users\maste\.nuget\packages\microsoft.windowsappsdk\1.2.230118.102\lib\net6.0-windows10.0.18362.0\Microsoft.Windows.ApplicationModel.DynamicDependency.Projection.dll
        C:\Users\maste\.nuget\packages\microsoft.windowsappsdk\1.2.230118.102\lib\net6.0-windows10.0.18362.0\Microsoft.Windows.ApplicationModel.Resources.Projection.dll
          Project file item includes which caused reference "C:\Users\maste\.nuget\packages\microsoft.windowsappsdk\1.2.230118.102\lib\net6.0-windows10.0.18362.0\Microsoft.Windows.ApplicationModel.Resources.Projection.dll".
            C:\Users\maste\.nuget\packages\microsoft.windowsappsdk\1.2.230118.102\lib\net6.0-windows10.0.18362.0\Microsoft.Windows.ApplicationModel.Resources.Projection.dll
        C:\Users\maste\.nuget\packages\microsoft.windowsappsdk\1.2.230118.102\lib\net6.0-windows10.0.18362.0\Microsoft.Windows.ApplicationModel.WindowsAppRuntime.Projection.dll
          Project file item includes which caused reference "C:\Users\maste\.nuget\packages\microsoft.windowsappsdk\1.2.230118.102\lib\net6.0-windows10.0.18362.0\Microsoft.Windows.ApplicationModel.WindowsAppRuntime.Projection.dll".
            C:\Users\maste\.nuget\packages\microsoft.windowsappsdk\1.2.230118.102\lib\net6.0-windows10.0.18362.0\Microsoft.Windows.ApplicationModel.WindowsAppRuntime.Projection.dll
        C:\Users\maste\.nuget\packages\microsoft.windowsappsdk\1.2.230118.102\lib\net6.0-windows10.0.18362.0\Microsoft.Windows.PushNotifications.Projection.dll
          Project file item includes which caused reference "C:\Users\maste\.nuget\packages\microsoft.windowsappsdk\1.2.230118.102\lib\net6.0-windows10.0.18362.0\Microsoft.Windows.PushNotifications.Projection.dll".
            C:\Users\maste\.nuget\packages\microsoft.windowsappsdk\1.2.230118.102\lib\net6.0-windows10.0.18362.0\Microsoft.Windows.PushNotifications.Projection.dll
        C:\Users\maste\.nuget\packages\microsoft.windowsappsdk\1.2.230118.102\lib\net6.0-windows10.0.18362.0\Microsoft.Windows.Security.AccessControl.Projection.dll
          Project file item includes which caused reference "C:\Users\maste\.nuget\packages\microsoft.windowsappsdk\1.2.230118.102\lib\net6.0-windows10.0.18362.0\Microsoft.Windows.Security.AccessControl.Projection.dll".
            C:\Users\maste\.nuget\packages\microsoft.windowsappsdk\1.2.230118.102\lib\net6.0-windows10.0.18362.0\Microsoft.Windows.Security.AccessControl.Projection.dll
        C:\Users\maste\.nuget\packages\microsoft.windowsappsdk\1.2.230118.102\lib\net6.0-windows10.0.18362.0\Microsoft.Windows.System.Power.Projection.dll
          Project file item includes which caused reference "C:\Users\maste\.nuget\packages\microsoft.windowsappsdk\1.2.230118.102\lib\net6.0-windows10.0.18362.0\Microsoft.Windows.System.Power.Projection.dll".
            C:\Users\maste\.nuget\packages\microsoft.windowsappsdk\1.2.230118.102\lib\net6.0-windows10.0.18362.0\Microsoft.Windows.System.Power.Projection.dll
        C:\Users\maste\.nuget\packages\microsoft.windowsappsdk\1.2.230118.102\lib\net6.0-windows10.0.18362.0\Microsoft.Windows.System.Projection.dll
          Project file item includes which caused reference "C:\Users\maste\.nuget\packages\microsoft.windowsappsdk\1.2.230118.102\lib\net6.0-windows10.0.18362.0\Microsoft.Windows.System.Projection.dll".
            C:\Users\maste\.nuget\packages\microsoft.windowsappsdk\1.2.230118.102\lib\net6.0-windows10.0.18362.0\Microsoft.Windows.System.Projection.dll
        C:\Users\maste\.nuget\packages\microsoft.windowsappsdk\1.2.230118.102\lib\net6.0-windows10.0.18362.0\Microsoft.Windows.Widgets.Projection.dll
          Project file item includes which caused reference "C:\Users\maste\.nuget\packages\microsoft.windowsappsdk\1.2.230118.102\lib\net6.0-windows10.0.18362.0\Microsoft.Windows.Widgets.Projection.dll".
            C:\Users\maste\.nuget\packages\microsoft.windowsappsdk\1.2.230118.102\lib\net6.0-windows10.0.18362.0\Microsoft.Windows.Widgets.Projection.dll	BugTracker	C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Current\Bin\amd64\Microsoft.Common.CurrentVersion.targets	2352	
Error	UXAML0001	An error was found in StackPanel	BugTracker	C:\Users\maste\Documents\GitHub\BugTracker\BugTracker\BugTracker\MainPage.xaml	20	
Error	UXAML0001	An error was found in Grid	BugTracker	C:\Users\maste\Documents\GitHub\BugTracker\BugTracker\BugTracker\MainPage.xaml	11	
Error	UXAML0001	An error was found in Page	BugTracker	C:\Users\maste\Documents\GitHub\BugTracker\BugTracker\BugTracker\MainPage.xaml	1	
Error	UXAML0001	Processing failed for file C:\Users\maste\Documents\GitHub\BugTracker\BugTracker\BugTracker\MainPage.xaml (Uno.UI.SourceGenerators.XamlGenerator.XamlParsingException: An error was found in Page ---> Uno.UI.SourceGenerators.XamlGenerator.XamlParsingException: An error was found in Grid ---> Uno.UI.SourceGenerators.XamlGenerator.XamlParsingException: An error was found in StackPanel ---> System.InvalidOperationException: Unable to find member [Type] on type [IssueItem]	BugTracker	C:\Users\maste\Documents\GitHub\BugTracker\BugTracker\BugTracker\CSC	1	
Error	UXAML0001	Unable to find member [Type] on type [IssueItem]	BugTracker	C:\Users\maste\Documents\GitHub\BugTracker\BugTracker\BugTracker\CSC	1	
Error	UXAML0001	An error was found in StackPanel	BugTracker	C:\Users\maste\Documents\GitHub\BugTracker\BugTracker\BugTracker\MainPage.xaml	20	
Error	UXAML0001	An error was found in Grid	BugTracker	C:\Users\maste\Documents\GitHub\BugTracker\BugTracker\BugTracker\MainPage.xaml	11	
Error	UXAML0001	An error was found in Page	BugTracker	C:\Users\maste\Documents\GitHub\BugTracker\BugTracker\BugTracker\MainPage.xaml	1	
Error	UXAML0001	Processing failed for file C:\Users\maste\Documents\GitHub\BugTracker\BugTracker\BugTracker\MainPage.xaml (Uno.UI.SourceGenerators.XamlGenerator.XamlParsingException: An error was found in Page ---> Uno.UI.SourceGenerators.XamlGenerator.XamlParsingException: An error was found in Grid ---> Uno.UI.SourceGenerators.XamlGenerator.XamlParsingException: An error was found in StackPanel ---> System.InvalidOperationException: Unable to find member [Type] on type [IssueItem]	BugTracker	C:\Users\maste\Documents\GitHub\BugTracker\BugTracker\BugTracker\CSC	1	
Error	UXAML0001	Unable to find member [Type] on type [IssueItem]	BugTracker	C:\Users\maste\Documents\GitHub\BugTracker\BugTracker\BugTracker\CSC	1	
Error	UXAML0001	An error was found in StackPanel	BugTracker	C:\Users\maste\Documents\GitHub\BugTracker\BugTracker\BugTracker\MainPage.xaml	20	
Error	UXAML0001	An error was found in Grid	BugTracker	C:\Users\maste\Documents\GitHub\BugTracker\BugTracker\BugTracker\MainPage.xaml	11	
Error	UXAML0001	An error was found in Page	BugTracker	C:\Users\maste\Documents\GitHub\BugTracker\BugTracker\BugTracker\MainPage.xaml	1	
Error	UXAML0001	Processing failed for file C:\Users\maste\Documents\GitHub\BugTracker\BugTracker\BugTracker\MainPage.xaml (Uno.UI.SourceGenerators.XamlGenerator.XamlParsingException: An error was found in Page ---> Uno.UI.SourceGenerators.XamlGenerator.XamlParsingException: An error was found in Grid ---> Uno.UI.SourceGenerators.XamlGenerator.XamlParsingException: An error was found in StackPanel ---> System.InvalidOperationException: Unable to find member [Type] on type [IssueItem]	BugTracker	C:\Users\maste\Documents\GitHub\BugTracker\BugTracker\BugTracker\CSC	1	
Error	UXAML0001	Unable to find member [Type] on type [IssueItem]	BugTracker	C:\Users\maste\Documents\GitHub\BugTracker\BugTracker\BugTracker\CSC	1	
Error	UXAML0001	An error was found in StackPanel	BugTracker	C:\Users\maste\Documents\GitHub\BugTracker\BugTracker\BugTracker\MainPage.xaml	20	
Error	UXAML0001	An error was found in Grid	BugTracker	C:\Users\maste\Documents\GitHub\BugTracker\BugTracker\BugTracker\MainPage.xaml	11	
Error	UXAML0001	An error was found in Page	BugTracker	C:\Users\maste\Documents\GitHub\BugTracker\BugTracker\BugTracker\MainPage.xaml	1	
Error	UXAML0001	Processing failed for file C:\Users\maste\Documents\GitHub\BugTracker\BugTracker\BugTracker\MainPage.xaml (Uno.UI.SourceGenerators.XamlGenerator.XamlParsingException: An error was found in Page ---> Uno.UI.SourceGenerators.XamlGenerator.XamlParsingException: An error was found in Grid ---> Uno.UI.SourceGenerators.XamlGenerator.XamlParsingException: An error was found in StackPanel ---> System.InvalidOperationException: Unable to find member [Type] on type [IssueItem]	BugTracker	C:\Users\maste\Documents\GitHub\BugTracker\BugTracker\BugTracker\CSC	1	
Warning	RAZORSDK1007	Reference assembly C:\Users\maste\Documents\GitHub\BugTracker\BugTracker\BugTracker\bin\Debug\net7.0\BugTracker.dll could not be found. This is typically caused by build errors in referenced projects.	BugTracker.Wasm	C:\Program Files\dotnet\sdk\7.0.102\Sdks\Microsoft.NET.Sdk.Razor\targets\Microsoft.NET.Sdk.Razor.MvcApplicationPartsDiscovery.targets	54	
Error	WMC1006	Cannot resolve Assembly or Windows Metadata file 'C:\Users\maste\Documents\GitHub\BugTracker\BugTracker\BugTracker\bin\Debug\net7.0-windows10.0.18362\BugTracker.dll'	BugTracker.Windows	C:\Users\maste\Documents\GitHub\BugTracker\BugTracker\BugTracker.Windows\BugTracker.Windows.csproj	1	

```
</details>

**Comment by sasakrsmanovic** [comment]
Those tutorials were designed so one goes through them in order. That said, we will look into the templates part you mention. 

Can you also confirm you've gone through the prerequisites page and Uno Check was run?

**Reply by elibroftw** [reply]
Yes, yes, and I was able to go through tutorial 1 without issues.

**Reply by jeromelaban** [reply]
Thanks for the report.
